CRM/Core/Config.php

```diff
@@ -575,4 +575,24 @@ public function handleFirstRun() {
     Civi::settings()->set('installed', 1);
   }
 
+  /**
+   * Is the system permitted to flush caches at the moment.
+   */
+  static public function isPermitCacheFlushMode() {
+    return !CRM_Core_Config::singleton()->doNotResetCache;
+  }
+
+  /**
+   * Set cache clearing to enabled or disabled.
+   *
+   * This might be enabled at the start of a long running process
+   * such as an import in order to delay clearing caches until the end.
+   *
+   * @param bool $enabled
+   *   If true then caches can be cleared at this time.
+   */
+  static public function setPermitCacheFlushMode($enabled) {
+    CRM_Core_Config::singleton()->doNotResetCache = $enabled ? 0 : 1;
+  }
```

Comment:
I'm generally a little confused about why `Config::$doNotResetCache` is being wrapped as a static function. I mean... technically... it's not changing anything, so there's no functional consequence either way. (Ex: `CRM_Core_Config::singleton(TRUE,TRUE)` still has the same consequence.)

Reply:
Is that idea that you're lining it up for some *other* change?

Reply:
Yep! that's the reason

Reply:
Personally I think it's more readable too.